src/main/java/reactor/netty/http/Http2StreamBridgeHandler.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2011-Present VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import io.netty.handler.codec.http.DefaultHttpContent;
+import io.netty.handler.codec.http2.Http2StreamFrameToHttpObjectCodec;
+
+import static reactor.netty.ReactorNetty.format;
+
+/**
+ * This handler is intended to work together with {@link Http2StreamFrameToHttpObjectCodec}
+ * it converts the outgoing messages into objects expected by
+ * {@link Http2StreamFrameToHttpObjectCodec}.
+ *
+ * @author Violeta Georgieva
+ * @since 1.0.0
+ */
+public class Http2StreamBridgeHandler extends ChannelDuplexHandler {
```

Comment:
Nit picking: Good reuse between client/server, Unfortunate it makes it in the public api tho. I wonder if separately you should think about using an "internal" subpackage for some of these or an annotation cc @rstoyanchev

Reply:
we will think about this in M2